library/network_connections.py

```diff
@@ -486,6 +486,31 @@ def ifcfg_create(
             rule4_file = content_current["rule"]
             rule6_file = content_current["rule6"]
 
+        if connection["802-1x"]:
+            ifcfg["IEEE_8021X_EAP_METHODS"] = str(connection["802-1x"]["eap"])
+            ifcfg["IEEE_8021X_IDENTITY"] = str(connection["802-1x"]["identity"])
+
+            ifcfg["IEEE_8021X_PRIVATE_KEY"] = str(
+                connection["802-1x"]["private-key-path"]
+            )
+            if connection["802-1x"]["private-key-password"]:
+                keys_file = "IEEE_8021X_PRIVATE_KEY_PASSWORD={}".format(
+                    str(connection["802-1x"]["private-key"]["password"])
+                )
+            if connection["802-1x"]["private-key-password-flags"]:
+                flags_str = Util.convert_passwd_flags_initscripts(
+                    connection["802-1x"]["private-key-password-flags"]
+                )
+                ifcfg["IEEE_8021X_PRIVATE_KEY_PASSWORD_FLAGS"] = flags_str
+
+            ifcfg["IEEE_8021X_CLIENT_CERT"] = str(connection["802-1x"]["client-cert"])
+
+            if connection["802-1x"]["ca-cert"]:
+                ifcfg["IEEE_8021X_CA_CERT"] = str(connection["802-1x"]["ca-cert"])
```

Comment:
when you do `grep -R IEEE_8021X_ /etc/sysconfig/`, you'll see that initscripts don't support any of these. See also, `/etc/sysconfig/network-scripts/ifup-wireless`.

The role also sets `NM_CONTROLLED=no`, so even if you have NetworkManager running, `ifup` won't call NetworkManager to handle this file. Hence, it's just not going to work.

At least, I doubt any of this will work. Did you test it? I would just drop initscripts support altogether and fail if running such a playbook against initscripts provider.

Reply:
Yep, my bad. Seems the 802.1x lines in initscripts are only used by network manager. I’m pretty sure you have to configure the 802.1x auth in wpa_supplicant.conf for it to work properly with NetworkManager disabled

Reply:
As far as testing, I only tested wireless 802.1x but obviously not very well!

I configured 802.1x with a wireless access point running OpenWRT, but do not have hardware that supports wired 802.1x.

Reply:
Not sure I understand. Do you mean, you also tested wireless 802.1x **with initscripts** and it worked? Interesting, I wouldn't expect that...

Reply:
I did test with initscripts and got it to work, but I believe my test method was flawed.
One of the tests I did was this:
* Configure wireless 802.1x using the role with "network_provider: nm"
* Ping the gateway
* Stop NetworkManager. Run ifdown
* Configure wireless 802.1x using the role with "network_provider: initscripts"
* Ping the gateway

I'm pretty sure what was happening was that NetworkManager set up the wpa_supplicant process correctly and this was still running in the background when I used initscripts, giving the impression that the initscripts method was working.

I had made the incorrect assumption that all parameters defined by NetworkManager in /etc/sysconfig/network-scripts were compatible with initscripts but this is clearly not the case.

Reply:
> I had made the incorrect assumption that all parameters defined by NetworkManager in /etc/sysconfig/network-scripts were compatible with initscripts but this is clearly not the case.

That is the aim in general, but only the case for a subset of parameters/options (also, because NetworkManager supports many more parameters than initscripts).